scripts/crypto.js

```diff
@@ -479,38 +419,31 @@ function getMarshaledBatchProof(proof) {
 }
 
 // Decode the proof that is sent into a map
-// 
+//
 // input: Marshaled proof string
 // output: DLEQ proof
 function parseDleqProof(proofStr) {
     const dleqProofM = JSON.parse(proofStr);
     let dleqProof = new Map();
-
-    // if we do not have the same commitments then something is wrong
-    if (!validateConsistentCommitments(dleqProofM.G, dleqProofM.H)) {
-        throw new Error(COMMITMENT_MISMATCH_ERR);
-    }
-
-    dleqProof.G = sec1DecodePoint(dleqProofM.G);
-    dleqProof.M = sec1DecodePoint(dleqProofM.M);
-    dleqProof.H = sec1DecodePoint(dleqProofM.H);
-    dleqProof.Z = sec1DecodePoint(dleqProofM.Z);
     dleqProof.R = getBigNumFromB64(dleqProofM.R);
     dleqProof.C = getBigNumFromB64(dleqProofM.C);
     return dleqProof;
 }
 
-// Check that the commitments on the proof match the commitments
-// in the extension
-function validateConsistentCommitments(G,H) {
-    if (G != activeCommConfig.G || H != activeCommConfig.H) {
-        return false;
-    }
-    return true;
-}
-
-// Return a byte array from a base-64 encoded string
+// Return a bignum from a base-64 encoded string
 function getBigNumFromB64(b64Str) {
-    let bits = sjcl.codec.base64.toBits(b64Str)
+    let bits = sjcl.codec.base64.toBits(b64Str);
     return sjcl.bn.fromBits(bits);
 }
+
+// Return a bignum from a hex string
+function getBigNumFromHex(hex) {
+    return sjcl.bn.fromBits(sjcl.codec.hex.toBits(hex));
+}
+
+// PRNG encode point
+function encodePointForPRNG(point) {
+    let hex = sjcl.codec.hex.fromBits(point.toBits());
+    let newHex = "04" + hex;
```

Comment:
is there anything special about this prefix?

Reply:
On the server-side all the points have this prefix. I'll make it static and add a comment explaining this though.

Reply:
The byte 0x04 means "uncompressed" for SEC1 marshaling. 0x02 or 0x03 would indicate compression with a particular sign (as seen [here](https://github.com/privacypass/challenge-bypass-extension/blob/master/scripts/crypto.js#L214) and [here](https://github.com/privacypass/challenge-bypass-server/blob/master/crypto/curve.go#L65))

Reply:
cool, assume this is the reference doc? http://www.secg.org/sec1-v2.pdf